### ISSUE TITLE
Underscore breaking social links in classic theme

### ISSUE THREAD

**Issue by nelas** [moderncv/moderncv#23]
Hello, thanks for maintaining the `moderncv` package. I was updating from an old version and noticed a small issue with the `\social` command.

Some of my social accounts have an underscore, and to be able to compile the document, I had to escape the `_` character like that:

```
\social[twitter]{my\_user}
```

However, this breaks the URL, which now points to `https://twitter.com/my/protect%20_user`

I solved it by changing the [`httpslink` command](https://github.com/moderncv/moderncv/blob/master/moderncv.cls#L535) to:

```
{\href{https://#2}{\detokenize{#1}}}}
```

And by not escaping the underscore:

```
\social[twitter]{my_user}
```

Not sure if that's the best solution, but it's working for now ;)

Cheers,
Bruno

**Comment by mefuller** [comment]
I just tested this now on a clean clone of the repo and I cannot reproduce this error.
The URL is not broken as you describe.
I compiled template.tex with and without the modified twitter handles (`jdoe` and `j\_doe`) and the links work flawlessly.

What TeX distribution and compiler are you using?
I used TeX-Live and XeLaTex on Ubuntu 20.04 (the template throws an error with pdflatex ("Encoding scheme `TU' unknown") that appears to be unrelated, otherwise I would have compiled with that.

Can you provide the files and information needed to reproduce this error (or check that it persists)?
Otherwise, I think this issue should be closed.

**Comment by nelas** [comment]
Hi @mefuller, thanks for looking into it. I realized the problem only occurs when using the `classic` theme:

```
\moderncvstyle{classic}
\social[twitter]{j\_doe}
```

It happens either with `pdflatex` or `xelatex` in a clean repository clone. Versions:

```
Ubuntu 20.10 Linux 5.8.0-33-generic
XeTeX 3.14159265-2.6-0.999992 (TeX Live 2020/Debian)
pdfTeX 3.14159265-2.6-1.40.21 (TeX Live 2020/Debian)
kpathsea version 6.3.2
Compiled with ICU version 67.1; using 67.1
Compiled with zlib version 1.2.11; using 1.2.11
Compiled with FreeType2 version 2.10.2; using 2.10.2
Compiled with Graphite2 version 1.3.14; using 1.3.14
Compiled with HarfBuzz version 2.6.4; using 2.6.4
Compiled with libpng version 1.6.37; using 1.6.37
Compiled with poppler version 0.68.0
Compiled with fontconfig version 2.13.1; using 2.13.1
Compiled with xpdf version 4.02
```

**Comment by mefuller** [comment]
I can confirm the bug and that your solution work.
The `detokenize` command seems pretty innocuous and nothing broke in my testing, so I created pull request #24 to address this.

**Comment by stephanlachnit** [comment]
Closed via #24.

**Comment by paddyroddy** [comment]
I'm using `banking` theme and recently noticed that twitter now includes the underscore
![image](https://user-images.githubusercontent.com/15052188/110255965-45ea5200-7f8e-11eb-990e-a2a119a3d84b.png)

**Comment by mefuller** [comment]
@paddyroddy are you reporting a bug?
The backslash escape character for underscores should no longer be necessary - your attached image appears to show the expected behavior based on pull #24

**Comment by paddyroddy** [comment]
> @paddyroddy are you reporting a bug?
> The backslash escape character for underscores should no longer be necessary - your attached image appears to show the expected behavior based on pull #24

ah, I hadn't realised the functionality changed, removing the `\` works as expected

**Comment by ethuleau** [comment]
I'm seeing a similar issue with special character like `é`:
I tried:
```latex
\social[linkedin]{joé-doe}
```
it compiles but it render a bad symbol `joÃľ-doe` with a bad link `https://www.linkedin.com/in/jo\Uffffffffoe`

I also tired:
```latex
\social[linkedin]{jo\'e-doe}  % optional, remove / comment the line if not wanted
```
it compiles but it does not render the special character `jo\’e-doe` with a bad link `https://www.linkedin.com/in/jo\Uffffffffoe`

Can we use special character in social link as Linkedin does?

**Comment by mefuller** [comment]
I can try investigating/testing this in the next couple days, but have you tried enclosing the special character in {}, e.g 
`\social[linkedin]{jo{é}-doe}` or `\social[linkedin]{jo{\'e}-doe}` ?
I don't know off-hand what's causing this and I'm also not sure how special characters are "supposed" to be rendered in URLs

**Comment by ethuleau** [comment]
I just tried and that renders the enclosing characters `{}` and the URl is not generated correctly `https://www.linkedin.com/in/jo{\Uffffffffdoe`

**Comment by paddyroddy** [comment]
I am running in to this issue again but where my `_` has been replaced with a floating `.`. I assume this is related to #79. The link works fine but should read `paddy_roddy`. Entered as `\social[twitter]{paddy_roddy}`

<img width="172" alt="image" src="https://user-images.githubusercontent.com/15052188/148931332-e910b61f-2ab7-4c11-ae39-481d42ebf1e5.png">

**Comment by paddyroddy** [comment]
> I am running in to this issue again but where my `_` has been replaced with a floating `.`. I assume this is related to #79. The link works fine but should read `paddy_roddy`. Entered as `\social[twitter]{paddy_roddy}`
> 
> <img alt="image" width="172" src="https://user-images.githubusercontent.com/15052188/148931332-e910b61f-2ab7-4c11-ae39-481d42ebf1e5.png">

changing to `lualatex` fixed this issue - copied in the latest changes from the template

**Comment by mefuller** [comment]
@paddyroddy Please open a new issue and post a minimal working example as well as indicate which compiler if you'd like us to look into this.
Without knowing more, I can't reproduce or investigate the bug.

**Comment by paddyroddy** [comment]
> @paddyroddy Please open a new issue and post a minimal working example as well as indicate which compiler if you'd like us to look into this. Without knowing more, I can't reproduce or investigate the bug.

I cannot recreate this since using the updated template as when compiling with `pdflatex` I get `! Undefined control sequence. \Url Error ->\url used in a moving argument.`

**Comment by mefuller** [comment]
we got some feedback yesterday about what I believe is a related issue (via our matrix channel)

1) the same error (`\Url Error ->\url used in a moving argument.`) is showing up only for the classic style (moderncvheadi.sty) (followed by a million other error messages)
2) compiler was Overleaf (which is by default pdfLaTeX - https://www.overleaf.com/learn/latex/Choosing_a_LaTeX_Compiler)

**Comment by mefuller** [comment]
tl;dr I found the smoking gun: the tikz definitions of missing icons in `moderncviconstikz.sty` cause compilation to fail when those icon definitions are loaded (occurs only when the icon is not previously defined by another set).
Still need to examine this in detail and decide how to proceed.
Help wanted.
--------------------------------------
I'm working with my "retokenized" files per #79 and have the following observations so far:

It looks like there are three social listings which may not be present in order to compile with pdfLaTeX: Signal, Matrix, and Google Scholar

with XeLaTeX, it's ok to put Google Scholar back in

Here's my analysis:
@ShadowMitia added the Google Scholar icon and cleaned up the social icons. I copied that work to add Signal, Matrix, and WhatsApp shortly after. Nothing in the way those are structured in `moderncv.cls` seems to be relevant - icons that don't cause errors are formatted identically to some that do, and it's not an issue (I think) of the total number of entries (I can't get Signal to work if I comment out other lines, for example).

Looking at the icon selections based on compiler, these three are all ones which are not present in `moderncviconsawesome.sty`, so I suspect that is the proximate cause of why pdfLaTeX is choking on them (see `moderncviconssymbols.sty'` (but the Researcher ID symbol is also not defined here and pdfLaTeX just outputs nothing for the icon instead of throwing errors).

The matrix and signal icons look to only be provided (and universally so) via the tikz definitions in `moderncviconstikz.sty`, from @gcorrad and this file doesn't contain a ResearcherID icon, so it looks like our compilation blows up when attempting to pull in an icon via the tikz definition. 
I commented out loading `moderncviconstikz.sty` in `moderncviconssymbols.sty' and the documents now compile with the full list of social information, just with no icons for Signal or Matrix showing at all and no Researcher ID or Google Scholar in pdfLaTeX.

**Comment by stephanlachnit** [comment]
Thank you so much for looking into this!

**Comment by mefuller** [comment]
Ah, but it just now occurs to me that the errors are only with classic, so something about that header mishandles the tikz icons.
I'll try to go through that in detail as compared to casual or banking.

**Comment by gcorrad** [comment]
It's because I forgotten to put \protect before tikz \path in social icons, I will open a PR so you can see... uff hard to find it.
Related to line 73 in moderncvheadi.sty (the head of classic)
Hope it works fine.
Thanks!

**Comment by gcorrad** [comment]
Sorry for the inconvenience

**Comment by mefuller** [comment]
@gcorrad No need to apologize - bugs happen and I'm glad we could work together to track it down

**Comment by gcorrad** [comment]
@mefuller Thank you!
